From d78ca42f6fbca923503f37e980c9e0c85dc96cc2 Mon Sep 17 00:00:00 2001
From: Richard Zhu <hongxing.zhu@nxp.com>
Date: Tue, 5 Jun 2018 16:45:29 +0800
Subject: [PATCH 3924/5242] MLK-17880 PCI: imx: fix the imx6q pcie resume hang

commit  5811c74c8350585f637990dca41bc522d1c715ef from
https://source.codeaurora.org/external/imx/linux-imx.git

The perst signal shouldn't be asserted during imx6q/dl
pcie suspend.
Otherwise, the link would be down during resume,
and system would be hang.
Fix it by one return in imx6q switch case.

Signed-off-by: Richard Zhu <hongxing.zhu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/pci/controller/dwc/pci-imx6.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/pci/controller/dwc/pci-imx6.c b/drivers/pci/controller/dwc/pci-imx6.c
index 988d5dd..3ade73d 100644
--- a/drivers/pci/controller/dwc/pci-imx6.c
+++ b/drivers/pci/controller/dwc/pci-imx6.c
@@ -1923,7 +1923,7 @@ static void pci_imx_pm_turn_off(struct imx_pcie *imx_pcie)
 		break;
 	case IMX6Q:
 		dev_info(dev, "Info: don't support pm_turn_off yet.\n");
-		break;
+		return;
 	}
 
 	udelay(1000);
-- 
1.7.9.5

